warp10/src/main/java/io/warp10/script/filler/FillerInterpolate.java

```diff
@@ -109,4 +115,111 @@ public int getPreWindow() {
   public int getPostWindow() {
     return 1;
   }
+  
+  public WarpScriptSingleValueFillerFunction compute(GeoTimeSerie gts) throws WarpScriptException {
+    
+    final PolynomialSplineFunction valFunction;
+    final PolynomialSplineFunction latFunction;
+    final PolynomialSplineFunction lonFunction;
+    final PolynomialSplineFunction elevationFunction;
+
+    if (gts.size() == 0) {
+      valFunction = null;
+      latFunction = null;
+      lonFunction = null;
+      elevationFunction = null;
+    } else {
+      if (GeoTimeSerie.TYPE.DOUBLE != gts.getType() && GeoTimeSerie.TYPE.LONG != gts.getType() && gts.size() > 0) {
+        throw new WarpScriptException(getName() + " expects a GTS of type DOUBLE or LONG, but instead got a GTS of type " + gts.getType().name());
```

Comment:
The message will be weird if an empty GTS whose type was forced is passed as parameter.

Reply:
The empty gts is managed a few lines earlier, I remove the condition.